src/pool.rs

```diff
@@ -141,20 +146,22 @@ impl ConnectionPool {
             if len < locked.size {
                 let socket = try!(self.connect());
                 let _ = locked.len.fetch_add(1, Ordering::SeqCst);
```

Comment:
should this now be incremented after the handshake, in case the handshake fails?

Reply:
Good catch!